selfdrive/test/longitudinal_maneuvers/plant.py

```diff
@@ -1,467 +1,158 @@
-  return msg2
-
-
-def car_plant(pos, speed, grade, gas, brake):
-  # vehicle parameters
-  mass = 1700
-  aero_cd = 0.3
-  force_peak = mass*3.
-  force_brake_peak = -mass*10.  # 1g
-  power_peak = 100000   # 100kW
-  speed_base = power_peak/force_peak
-  rolling_res = 0.01
-  gravity = 9.81
-  frontal_area = 2.2
-  air_density = 1.225
-  gas_to_peak_linear_slope = 3.33
-  brake_to_peak_linear_slope = 0.3
-  creep_accel_v = [1., 0.]
-  creep_accel_bp = [0., 1.5]
-
-  #*** longitudinal model ***
-  # find speed where peak torque meets peak power
-  force_brake = brake * force_brake_peak * brake_to_peak_linear_slope
-  if speed < speed_base:  # torque control
-    force_gas = gas * force_peak * gas_to_peak_linear_slope
-  else:  # power control
-    force_gas = gas * power_peak / speed * gas_to_peak_linear_slope
-
-  force_grade = - np.sin(np.arctan(grade)) * mass * gravity
-
-  creep_accel = np.interp(speed, creep_accel_bp, creep_accel_v)
-  force_creep = creep_accel * mass
-
-  force_resistance = -(rolling_res * mass * gravity + 0.5 * speed**2 * aero_cd * air_density * frontal_area)
-  force = force_gas + force_brake + force_resistance + force_grade + force_creep
-  acceleration = force / mass
-
-  # TODO: lateral model
-  return speed, acceleration
-
-def get_car_can_parser():
-  dbc_f = 'honda_civic_touring_2016_can_generated'
-  signals = [
-    ("STEER_TORQUE", 0xe4, 0),
-    ("STEER_TORQUE_REQUEST", 0xe4, 0),
-    ("COMPUTER_BRAKE", 0x1fa, 0),
-    ("COMPUTER_BRAKE_REQUEST", 0x1fa, 0),
-    ("GAS_COMMAND", 0x200, 0),
-  ]
-  checks = [
-    (0xe4, 100),
-    (0x1fa, 50),
-    (0x200, 50),
-  ]
-  return CANParser(dbc_f, signals, checks, 0)
-
-def to_3_byte(x):
-  # Convert into 12 bit value
-  s = struct.pack("!H", int(x))
-  return binascii.hexlify(s)[1:]
-
-def to_3s_byte(x):
-  s = struct.pack("!h", int(x))
-  return binascii.hexlify(s)[1:]
 
 class Plant():
   messaging_initialized = False
 
-  def __init__(self, lead_relevancy=False, rate=100, speed=0.0, distance_lead=2.0):
+  def __init__(self, lead_relevancy=False, rate=20, speed=0.0, distance_lead=2.0):
     self.rate = rate
 
     if not Plant.messaging_initialized:
-      Plant.pm = messaging.PubMaster(['roadCameraState', 'driverCameraState', 'ubloxRaw'])
-      Plant.logcan = messaging.pub_sock('can')
-      Plant.sendcan = messaging.sub_sock('sendcan')
-      Plant.model = messaging.pub_sock('modelV2')
-      Plant.live_params = messaging.pub_sock('liveParameters')
-      Plant.live_location_kalman = messaging.pub_sock('liveLocationKalman')
-      Plant.pandaState = messaging.pub_sock('pandaState')
-      Plant.deviceState = messaging.pub_sock('deviceState')
-      Plant.driverMonitoringState = messaging.pub_sock('driverMonitoringState')
-      Plant.cal = messaging.pub_sock('liveCalibration')
-      Plant.controls_state = messaging.sub_sock('controlsState')
+      Plant.radar = messaging.pub_sock('radarState')
```

Comment:
ideally this shouldn't use any real sockets. should be able to reuse process replay's fake pub and sub master

Reply:
we can do this later